src/flowcean/polars/transforms/resample_to_reference.py

```diff
@@ -0,0 +1,164 @@
+    ref = (
+        data.with_row_index()
+        .explode(reference)
+        .select(
+            pl.col("index"),
+            pl.col(reference).struct.field("time").alias("time"),
+        )
+    )
+
+    # For each feature: explode → forward-fill onto reference timeline
+    aligned_cols = []
+    for col in features:
+        df = (
+            data.with_row_index()
+            .explode(col)
+            .select(
+                pl.col("index"),
+                pl.col(col).struct.field("time").alias("t"),
+                pl.col(col)
+                .struct.field("value")
+                .name.prefix_fields(f"{col}/")
+                .struct.unnest(),
+            )
+        )
+
+        # Join to reference timeline (as-of join)
+        df = ref.join_asof(
+            df.sort("t").drop("index"),
+            left_on="time",
+            right_on="t",
+        ).sort("index")
+
+        # Forward-fill the values along the reference sampling times
+        df = df.with_columns(pl.exclude("index", "time", "t").forward_fill())
+
+        aligned_cols.append(df)
+
+    final = aligned_cols[0]
+    for df in aligned_cols[1:]:
+        feature_cols = [
+            c
+            for c in df.collect_schema().names()
+            if c not in ["index", "time", "t"]
+        ]
+        final = pl.concat([final, df.select(feature_cols)], how="horizontal")
+
+    # Group by index and aggregate into time series
+    return (
+        final.select(
+            pl.col("index"),
+            pl.struct(
+                pl.col("time"),
+                pl.struct(pl.exclude("index", "time", "t")).alias("value"),
+            ).alias(name),
+        )
+        .group_by("index", maintain_order=True)
+        .agg(pl.all().implode())
+        .drop("index")
+    )
+
+
+class ResampleToReference(Transform):
+    """Resamples time series features to a reference timeline.
+
+    This transform takes multiple time series features and resamples them all
+    to match the sampling times of a specified reference feature. Values are
+    forward-filled (holding the last value) to the reference timestamps.
+
+    Example:
+        If reference has timestamps [0, 5, 10] and another feature has
+        timestamps [0, 3, 7, 12], the output will have timestamps [0, 5, 10]
+        with the other feature's values forward-filled to those times.
+    """
+
+    def __init__(
+        self,
+        reference: str,
+        features: list[str] | None = None,
+        name: str = "resampled",
+        *,
+        drop: bool = True,
+    ) -> None:
+        """Initialize the ResampleToReference transform.
+
+        Args:
+            reference: Column whose timestamps define the output timeline.
+                This column must be included in features (or present in data if
+                features is None).
+            features: List of time series features to resample. If None, uses
+                all features in the data. Must include reference.
```

Comment:
Why does the reference column be included in the features? Depending on interpretation this is either obvious, or not intended and should be optional